packages/gafl-webapp-service/src/pages/order-complete/licence-details/__tests__/route.spec.js

```diff
@@ -26,149 +25,166 @@ jest.mock('../../../../constants.js', () => ({
-      const mockRequest = createMockRequest({ i18nCatalog: { [catKey]: ageConcession } })
-      const { ageConcessionText } = await getData(mockRequest)
-      expect(ageConcessionText).toBe(ageConcession)
+    it('getAgeConcession is called with permission', async () => {
+      const mockPermission = getMockPermission()
+      const mockRequest = getMockRequest(getMockStatus(), getMockTransaction(mockPermission))
+      await getData(mockRequest)
+      expect(concessionHelper.getAgeConcession).toHaveBeenCalledWith(mockPermission)
     })
 
-    it('returns the expected data', async () => {
-      const mockRequest = createMockRequest()
-      dtDisplay.displayStartTime.mockReturnValueOnce('1:00am on 6 June 2020')
-      dtDisplay.displayEndTime.mockReturnValueOnce('1:00am on 5 June 2021')
-      licenceTypeDisplay.mockReturnValueOnce('Shopping trollies and old wellies')
-      concessionHelper.hasDisabled.mockReturnValueOnce(true)
-      concessionHelper.hasSenior.mockReturnValueOnce(true)
-      concessionHelper.getAgeConcession.mockReturnValueOnce({ proof: { type: 'No Proof' }, type: 'Senior' })
-      const result = await getData(mockRequest)
-      expect(result).toMatchSnapshot()
+    describe('returns expected licence data for the given permission:', () => {
+      licenceTypeDisplay.mockReturnValue('Licence Type')
+      displayStartTime.mockReturnValue('Start Time')
+      displayEndTime.mockReturnValue('End Time')
+      concessionHelper.hasDisabled.mockReturnValue('Disability')
+
+      it('returns expected data', async () => {
+        const data = await getData(getMockRequest(getMockStatus(), getMockTransaction()))
+        expect(data).toMatchSnapshot()
+      })
+
+      it.each([['1D'], ['8D'], ['12M']])('returns licence length as %s', async licenceLength => {
+        licenceTypeAndLengthDisplay.mockReturnValue(licenceLength)
+        const data = await getData(getMockRequest(getMockStatus(), getMockTransaction()))
+        expect(data).toMatchSnapshot()
+      })
+
+      it.each([['Senior'], ['Junior'], ['Neither']])('returns age concession as %s', async type => {
+        const concession = {
+          type: type
+        }
+        concessionHelper.getAgeConcession.mockReturnValue(concession)
+        const data = await getData(getMockRequest(getMockStatus(), getMockTransaction()))
+        expect(data).toMatchSnapshot()
+      })
     })
-  })
-})
 
-const getSamplePermission = () => ({
-  startDate: '2020-06-06',
-  endDate: '2021-06-05',
-  licenceType: LICENCE_TYPE['trout-and-coarse'],
-  numberOfRods: '3',
-  concessions: [
-    {
-      type: CONCESSION.DISABLED,
-      proof: {
-        type: CONCESSION_PROOF.blueBadge,
-        referenceNumber: '123456324'
-      }
-    },
-    {
-      type: CONCESSION.SENIOR,
-      proof: {
-        type: CONCESSION_PROOF.none
-      }
-    }
-  ]
-})
+    describe('throws a Boom forbidden error', () => {
+      it('boom is called', async () => {
```

Comment:
We just need a bit better description here. Why is boom called? What is different about the data that means an error is thrown?